### PR TITLE
Channel-type-crud

### PR DESCRIPTION
**Pull request by peterdeme**
- Channel type CRUD commands

### CODE REVIEW

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+		updateCmd(),
+		listCmd()}
+}
+
+func getCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-channel-type --channel-type [channel-type] --output-format [json]",
+		Short: "Get channel type",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetStreamClient(cmd)
```
Comment (ferhatelmas) [inline pkg/cmd/chat/channeltype/channeltype.go:28]: slowly drop prefix such that `GetClient`

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+				unindented, err := json.Marshal(resp)
+				if err != nil {
+					return err
+				}
+
+				var indented bytes.Buffer
+				err = json.Indent(&indented, unindented, "", "  ")
+				if err != nil {
+					return err
+				}
```
Comment (ferhatelmas) [inline pkg/cmd/chat/channeltype/channeltype.go:51]: might want to use https://pkg.go.dev/encoding/json#NewEncoder

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+				cmd.Println(indented.String())
+			} else {
+				return fmt.Errorf("unknown output format: %s", format)
```
Comment (ferhatelmas) [inline pkg/cmd/chat/channeltype/channeltype.go:55]: good to handle it first

**Review by ferhatelmas — COMMENTED** [review]
a few small bits, overall fine

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+			}
+
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("channel-type", "t", "", "Channel type")
```
Comment (yaziine) [inline pkg/cmd/chat/channeltype/channeltype.go:63]: `type` please

File: pkg/cmd/chat/channeltype/channeltype.go (hunk 1 above)
Comment (peterdeme) [reply pkg/cmd/chat/channeltype/channeltype.go:28]: ✅

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+			}
+
+			format, _ := cmd.Flags().GetString("output-format")
+
+			resp, err := c.ListChannelTypes(cmd.Context())
+			if err != nil {
+				return err
+			}
+
+			if format == "json" {
```
[the remark this answers is not in the document]
Comment (peterdeme) [reply pkg/cmd/chat/channeltype/channeltype.go:189]: ✅

File: pkg/cmd/chat/channeltype/channeltype.go (hunk 4 above)
Comment (peterdeme) [reply pkg/cmd/chat/channeltype/channeltype.go:63]: ✅

**Review by ferhatelmas — APPROVED** [review]
LGTM

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+		getCmd(),
+		createCmd(),
+		deleteCmd(),
+		updateCmd(),
+		listCmd()}
+}
+
+func getCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-channel-type --channel-type [channel-type] --output-format [json]",
```
Comment (yaziine) [inline pkg/cmd/chat/channeltype/channeltype.go:24]:
```suggestion
		Use:   "get-channel-type --type [channel type] --output-format [json]",
```

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	fl := cmd.Flags()
+	fl.StringP("properties", "p", "", "Raw JSON properties")
+	cmd.MarkFlagRequired("properties")
+
+	return cmd
+}
+
+func deleteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete-channel-type --channel-type [channel-type]",
```
Comment (yaziine) [inline pkg/cmd/chat/channeltype/channeltype.go:88]:
```suggestion
		Use:   "delete-channel-type --type [channel type]",
```

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+	fl := cmd.Flags()
+	fl.StringP("type", "t", "", "Channel type")
+	cmd.MarkFlagRequired("type")
+
+	return cmd
+}
+
+func updateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-channel-type --channel-type [channel-type] --properties [raw-json-properties]",
```
Comment (yaziine) [inline pkg/cmd/chat/channeltype/channeltype.go:117]:
```suggestion
		Use:   "update-channel-type --type [channel-type] --properties [raw-json-properties]",
```

File: pkg/cmd/chat/channeltype/channeltype.go [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+	fl.StringP("properties", "p", "", "Raw JSON properties")
+	cmd.MarkFlagRequired("type")
+	cmd.MarkFlagRequired("properties")
+
+	return cmd
+}
+
+func listCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-channel-types --channel-type [channel-type] --output-format [json]",
```
Comment (yaziine) [inline pkg/cmd/chat/channeltype/channeltype.go:155]:
```suggestion
		Use:   "list-channel-types --output-format [json]",
```

File: pkg/utils/printer.go [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+		return err
+	}
+
+	switch format {
+	case "json":
+		return printJsonObject(cmd, object)
+	default:
+		return fmt.Errorf("unknown output format: %s", format)
+	}
+
```
Comment (yaziine) [inline pkg/utils/printer.go:23]:
```suggestion
```

**Pull request merged**